File: elemi_programozasi_tetelek/osszegzes/osszegzes.cpp
#include <iostream>
#include <vector>

using namespace std;

double Osszeg1(vector<double> lista)
{
    double sum = 0;
    for (double& elem : lista)
    {
        sum += elem;
    }
    return sum;
}

double Osszeg2(vector<double> lista)
{
    double sum = 0;
    for (int i = 0; i < lista.size(); i++)
    {
        sum += lista[i];
    }
    return sum;
}

int main()
{
    vector<double> l = { 1, 2, 3, 4, 5 };
    cout << Osszeg1(l) << endl << Osszeg2(l);
}

File: elemi_programozasi_tetelek_rekurzivan/osszegzes/osszegzes.js
function osszeg1(lista){
    let sum = 0;
    for (const elem of lista) {
        sum += elem;
    }
    return sum;
}

function osszeg2(lista){
    let sum = 0;
    for (let i = 0; i < lista.length; i++) {
        sum += lista[i];
    }
    return sum;
}

let l = [1, 2, 3, 4, 5];
console.log(osszeg1(l));
console.log(osszeg2(l));
